server/main.py

```diff
@@ -153,6 +160,111 @@ async def game_loop():
+                )
+            case {"player": uid, "direction": direction, "success": success}:
+                player_uids = uid
+                succeed = "succeed!" if success else "fail!"
+                update = ActionUpdateMessage(
+                    type="update",
+                    message=f"You try moving {direction}... and {succeed}",
+                )
+            case {"type": "room_action"}:
+                player_uids = get_room_update_uids(action)
+                update = ActionUpdateMessage(
+                    type="update", message=get_room_update_message(action)
+                )
+            case int():
+                player_uids = action
+                update = ActionUpdateMessage(
+                    type="update",
+                    message="Time passes by, but you didn't do anything this round!",
+                )
+
+        match player_uids:
+            case int():
+                # Send to a single player.
+                player_connection = connections.get(player_uids)
+                if player_connection is not None:
+                    await player_connection.send(update.json())
+                # else: player disconnected.
+            case set():
+                # Broadcast to multiple players.
+                player_connections = {
+                    connections.get(player_uid)
+                    for player_uid in player_uids
+                    if connections.get(player_uid) is not None
+                }
+                websockets.broadcast(player_connections, update.json())
+
+
+def get_action_update_message(action: ActionDict) -> str:
+    """
+    This is the message to be displayed to the player that cast this action.
+
+    A big mess to get a proper message for the action!
+    """
+    target = game.get_mob(action["target"]) or game.get_player(action["target"])
+    target_name = "yourself" if action["target"] == action["caster"] else target.name
+    assert target
+
+    tried = "attack" if action["dmg"] > 0 else "heal"
+    health_affected = abs(action["dmg"])
+
+    if not action["cast"]:
+        message = f"You tried {tried}ing {target_name} but you don't have mana!"
+    else:
+        if action["hit"]:
+            message = f"You {tried} for {health_affected} hit points!"
+        else:
+            message = f"You tried {tried}ing {target_name} but missed!"
+
+    return message
+
+
+def get_room_update_uids(room_action: RoomActionDict) -> set:
+    """Returns the uids of every player in the room except the caster."""
+    players = room_action["room"].get_players()
+    uids = {
+        player.uid
+        for player in players
+        if player.uid != room_action["action"]["caster"]
+    }
+
+    return uids
+
+
+def get_room_update_message(room_action: RoomActionDict) -> str:
+    """This is the message to be displayed to the players in the action that this room happened."""
+    RoomActionDict()
```

Comment:
As far as I can tell, this doesn't have any effect. What's it for?

Reply:
it doesn't, I put it there because it helped me with some type hints and forgot to remove it :PPP